heare/developer/compacter.py

```diff
@@ -188,17 +271,24 @@ def _estimate_token_count(self, text: str) -> int:
         words = len(text.split())
         return int(words / 0.75)
 
-    def should_compact(self, messages: List[MessageParam], model: str) -> bool:
+    def should_compact(self, messages: List[MessageParam], model: str, context_dict: dict = None) -> bool:
```

Comment:
change the method signature here to just take an AgentContext instance. Update all the callsites.